### DISCUSSION TITLE
Globally adding a newly generated language dictionary

### DISCUSSION THREAD

**Discussion by ttasovac** [streetsidesoftware/vscode-spell-checker#2420] [Q&A] [answered]
I've converted my Serbian dictionary from Hunspell to cspell using the cspell-cli tools. I have added it to cspell with `cspell link add ./cspell-ext.json`  and I can trace Serbian words from the command line, i.e. cspell is aware of my Serbian dictionary.

In `@ext:streetsidesoftware.code-spell-checker` I've added `sr-RS` as an item. I've also added this:

```jsonc
"cSpell.dictionaryDefinitions": [
        {
            // The name of the dictionary is used to look it up.
            "name": "sr-RS",
            // Path to the custom word file. Relative to this `cspell.json` file.
            "path": "/Users/ttasovac/Development/ttasovac-forks/cspell-dicts/dictionaries/sr_RS/sr_rs.trie.gz",
            // Some editor extensions will use `addWords` for adding words to your
            // personal dictionary.
            "addWords": true
        }
    ]
```

to the user `settings.json` but vscode-spell-checker still can't find my dictionary.

![Снимок экрана 2022-12-30 в 13 09 31](https://user-images.githubusercontent.com/5304839/210069224-a82035e9-a412-40f3-aeae-6ec12c0517a9.png)

I'm pretty sure this is documented somewhere, but I'm completely new to VSCode —  today is my second day using it —  and I'm a bit overwhelmed by all those different configuration options. I'd be most grateful for any tips.

**Comment by Jason3S** [comment, marked as answer]
@ttasovac,

VSCode doesn't deal with "array" settings very well.

Please try using `cSpell.customDictionaries` instead.

<img width="984" alt="image" src="https://user-images.githubusercontent.com/3740137/210078244-5db13941-efa5-4dde-9551-f61fee584ee6.png">

Example:

```json
"cSpell.customDictionaries": {
    "c-words": {
        "name": "c-words",
        "path": "~/cwords/all.txt"
    }
},
```

Do not set `addWords` to `true`. It is not possible to add words to a `trie.gz` file.

**Comment by ttasovac** [comment]
Thanks a lot, @Jason3S —  this was very helpful.

I can now use the custom Serbian dictionary with this:  

```json
  "cSpell.customDictionaries": {
    "sr-RS": {
      "name": "sr-RS",
      "path": "/Users/ttasovac/Development/ttasovac-forks/cspell-dicts/dictionaries/sr_RS/sr_rs.trie.gz",
      "description": "Serbian dictionary",
      "addWords": false,
      "scope": "user"
    }
  },
```

I'm still wondering however, why I'm getting this: 

<img width="726" alt="Снимок экрана 2023-01-02 в 06 18 07" src="https://user-images.githubusercontent.com/5304839/210197297-e0bdec56-9fdc-4299-a8a9-c857cdae57f4.png">

 and this:

<img width="730" alt="Снимок экрана 2023-01-02 в 06 18 19" src="https://user-images.githubusercontent.com/5304839/210197355-3c4ead4d-980a-4a8d-8658-f131cb370582.png">

How does the extension pick up the correct name, icon and locales here? 

Many thanks in advance.

**Reply by Jason3S** [reply]
@ttasovac,

It calculates that from the `langaugeSettings`.

<img width="714" alt="image" src="https://user-images.githubusercontent.com/3740137/210263927-d5251ba4-cc45-4ff1-b652-8a5018c7e3f0.png">

```json
"cSpell.languageSettings": [
    {
        "languageId": "*",
        "locale": "sr,sr-RS",
        "dictionaries": ["sr-RS"]
    }
]
```

The spell checker uses `languageSettings` to bind the locale to the dictionary.

**Comment by ttasovac** [comment]
Thanks a lot, @Jason3S!
